### PR TITLE
feat: pyvw label accessor for continuous actions

### PR DESCRIPTION
**Pull request by bassmang**
Addresses #3043

### CODE REVIEW

File: python/pylibvw.cc [hunk 1]
```diff
@@ -864,6 +864,11 @@ uint32_t ex_get_cbandits_class(example_ptr ec, uint32_t i) { return ec->l.cb.cos
 float ex_get_cbandits_probability(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].probability; }
 float ex_get_cbandits_partial_prediction(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].partial_prediction; }
 
+uint32_t ex_get_cb_continuous_num_costs(example_ptr ec) { return (uint32_t)ec->l.cb_cont.costs.size(); }
+float ex_get_cb_continuous_cost(example_ptr ec, uint32_t i) { return ec->l.cb_cont.costs[i].cost; }
```
Comment (jackgerrits) [inline python/pylibvw.cc:868]: Bounds check and throw?

File: python/vowpalwabbit/pyvw.py [hunk 2]
```diff
@@ -1184,6 +1184,43 @@ def __str__(self):
         )
 
 
+class cb_continuous_label(abstract_label):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1187]: We should conform to PEP8 for newly added code and move towards it for old code
```suggestion
class CBContinuousLabel(abstract_label):
```

File: python/vowpalwabbit/pyvw.py [hunk 3]
```diff
@@ -1184,6 +1184,43 @@ def __str__(self):
...
+    """Class for cb_continuous VW label"""
+
+    def __init__(self, costs=[]):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1190]: Add type

File: python/vowpalwabbit/pyvw.py [hunk 4]
```diff
@@ -1184,6 +1184,43 @@ def __str__(self):
...
+        abstract_label.__init__(self)
+        if isinstance(costs, example):
+            self.from_example(costs)
+        else:
+            self.costs = costs
+
+    def from_example(self, ex):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1197]: add type

File: python/vowpalwabbit/pyvw.py [hunk 5]
```diff
@@ -1184,6 +1184,43 @@ def __str__(self):
...
 
+class cb_continuous_label(abstract_label):
+    """Class for cb_continuous VW label"""
+
+    def __init__(self, costs=[]):
+        abstract_label.__init__(self)
+        if isinstance(costs, example):
+            self.from_example(costs)
+        else:
+            self.costs = costs
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1195]: Add a type definition for the class

File: python/pylibvw.cc (hunk 1 above)
Comment (bassmang) [reply python/pylibvw.cc:868]: Resolved comments and added similar changes for `cb`

File: python/vowpalwabbit/pyvw.py [hunk 6]
```diff
@@ -1134,86 +1134,83 @@ def __str__(self):
...
+                DeprecationWarning,
+            )
+        self.label = action
+        self.action = action
+        self.cost = cost
+        self.partial_prediction = partial_prediction
+        self.probability = probability
+
+
+class CBandits_label(abstract_label):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1154]: This is a potential breaking change so we should change the name but add a deprecated way to use the old name

File: python/vowpalwabbit/pyvw.py [hunk 7]
```diff
@@ -1134,86 +1134,83 @@ def __str__(self):
...
+        action=None,
+        cost=0.0,
+        pdf_value=0.0,
+    ):
+        self.action = action
+        self.cost = cost
+        self.pdf_value = pdf_value
+
+
+class CBContinuous_label(abstract_label):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1195]:
```suggestion
class CBContinuousLabel(abstract_label):
```

File: python/vowpalwabbit/pyvw.py (hunk 6 above)
Comment (jackgerrits) [reply python/vowpalwabbit/pyvw.py:1154]: PEP8 states classes should use PascalCase

Comment (bassmang) [reply python/vowpalwabbit/pyvw.py:1154]: I'll add this in a separate PR with some other classes

Comment (jackgerrits) [reply python/vowpalwabbit/pyvw.py:1154]: Okay, best to avoid changing this name in this PR then

File: python/vowpalwabbit/pyvw.py [hunk 8]
```diff
@@ -1134,56 +1134,90 @@ def __str__(self):
...
-            self.costs.append(wc)
+            self.costs.append(cb)
 
     def __str__(self):
         return " ".join(
             ["{}:{}:{}".format(c.action, c.cost, c.probability) for c in self.costs]
         )
 
 
+class continuous_label_elm:
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1183]: ContinuousLabelElement

File: python/vowpalwabbit/pyvw.py [hunk 9]
```diff
@@ -1134,56 +1134,90 @@ def __str__(self):
...
-            self.costs.append(wc)
+            self.costs.append(cb)
 
     def __str__(self):
         return " ".join(
             ["{}:{}:{}".format(c.action, c.cost, c.probability) for c in self.costs]
         )
 
 
+class ContinuousLabelElement:
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1183]: Can you add class types?

```
class ContinuousLabelElement:
    cost: float
```

Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1183]: To be consistent should this be `CBContinuousLabelElement`?

File: python/vowpalwabbit/pyvw.py [hunk 10]
```diff
@@ -1134,56 +1134,90 @@ def __str__(self):
...
+            self.costs.append(cb)
 
     def __str__(self):
         return " ".join(
             ["{}:{}:{}".format(c.action, c.cost, c.probability) for c in self.costs]
         )
 
 
+class CBContinuousLabelElement:
+    def __init__(self, action: int = None, cost: float = 0.0, pdf_value: float = 0.0):
```
Comment (jackgerrits) [inline python/vowpalwabbit/pyvw.py:1189]:
```suggestion
    def __init__(self, action: Optional[int] = None, cost: float = 0.0, pdf_value: float = 0.0):
```

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**